src/server/scripts/Kalimdor/CavernsOfTime/BattleForMountHyjal/instance_hyjal.cpp

```diff
@@ -37,7 +30,48 @@ EndScriptData */
 4 - Archimonde event
 */
 
-#define YELL_EFFORTS        "All of your efforts have been in vain, for the draining of the World Tree has already begun. Soon the heart of your world will beat no more."
+DoorData const doorData[] =
+{
+    { GO_HORDE_ENCAMPMENT_PORTAL,  DATA_ANETHERON,  DOOR_TYPE_PASSAGE },
+    { GO_NIGHT_ELF_VILLAGE_PORTAL, DATA_AZGALOR,    DOOR_TYPE_PASSAGE }
```

Comment:
Missing "end of array" item in the list.

Reply:
from what I've seen so far DoorData doesn't use it. Gives me an error if I try as well.

Reply:
DoorData needs to have the last entry set to 0. Otherwise, it could lead to undefined behavior.
```suggestion
    { GO_NIGHT_ELF_VILLAGE_PORTAL, DATA_AZGALOR,    DOOR_TYPE_PASSAGE },
    { 0,                           0,               DOOR_TYPE_PASSAGE }
```

Reply:
I did do that in TK. But also SSC doesn't have it and it seems to work fine (but that says nothing ofc). Overall it's better to do it than not to do it as it does seem to work fine for TK

Reply:
I used SSC as a template lol, that's why.
Ok, I didn't know about this, ty @walkline